File: Source/Fantasy.Jobs/Resources/IResourceProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Fantasy.Jobs.Resources
{
    public interface IResourceProvider
    {
        bool CanHandle(string name);

        void Initialize();

        bool IsAvailable(ResourceParameter parameter);

        bool Request(ResourceParameter parameter, out object resource);

        void Release(object resource);

        event EventHandler Available;

        event EventHandler<ProviderRevokeArgs> Revoke;
    }
}

File: Fantasy.Jobs/src/fantasy/jobs/resources/IResourceProvider.java
package fantasy.jobs.resources;

import fantasy.*;

public interface IResourceProvider
{
	boolean canHandle(String name)  throws Exception;

	void initialize()  throws Exception;

	boolean isAvailable(ResourceParameter parameter)  throws Exception;

	boolean request(ResourceParameter parameter, RefObject<Object> resource)  throws Exception;

	void release(Object resource) throws Exception;
	
	void addListener(IResourceProviderListener listener);
	void removeListener(IResourceProviderListener listener);

}